bookkeeper-server/src/main/java/org/apache/bookkeeper/client/LedgerHandle.java

```diff
@@ -791,6 +801,17 @@ public void onFailure(Throwable cause) {
                                                               long lastEntry) {
         PendingReadOp op = new PendingReadOp(this, bk.getScheduler(), firstEntry, lastEntry);
         if (!bk.isClosed()) {
+            // waiting on the first one.
+            // not very helpful if there are multiple ensembles or if bookie goes into unresponsive
+            // state later after N requests sent.
+            // unfortunately it seems that alternatives are:
+            // - send reads one-by-one (our case anyway)
+            // - block worker pool (not good)
+            DistributionSchedule.WriteSet ws = distributionSchedule.getWriteSet(firstEntry);
```

Comment:
`WriteSet` is a recycle object, you need to recycle it after usage.

Reply:
good catch, thank you.